### PR TITLE
Created three compontents out of detail view component 

### PR DESCRIPTION
**Pull request by pravagar**
Created three components out of detail view component and used those to render detail page.

-Used JavaScript way of styling

### CODE REVIEW

File: ServerlessLibraryAPI/ClientApp/src/components/DetailView/DetailPageContent.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+      selectedKey: pivotItem.props.itemKey
+    });
+    if (
+      pivotItem.props.itemKey === "armtemplate" &&
+      this.state.armTemplateText === ""
+    ) {
+      let { template } = this.props;
+      if (template) {
+        fetch(template)
+          .then(response => response.text())
```
Comment (msnehagup) [inline ServerlessLibraryAPI/ClientApp/src/components/DetailView/DetailPageContent.js]: you can add handling for text responses in handle-response.js -> we can build that helper to handle negative scenarios properly in a single place

File: ServerlessLibraryAPI/ClientApp/src/components/DetailView/DetailPageHeader.scss [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  display: flex;
+  flex-direction: row;
+  margin-bottom: 11px;
+}
+
+.back-button-icon-container {
+  margin: auto 0;
+}
+
+.backButtonStyles {
```
Comment (msnehagup) [inline ServerlessLibraryAPI/ClientApp/src/components/DetailView/DetailPageHeader.scss]: same styling is also defined in the js file. is it needed in both places?

File: ServerlessLibraryAPI/ClientApp/src/components/DetailView/DetailView.js [hunk 3]
```diff
@@ -1,58 +1,18 @@
...
-      selectedKey: "licence"
+      sample: {}
     };
-
-    this.handleBackButtonClick = this.handleBackButtonClick.bind(this);
-    this.handleLinkClick = this.handleLinkClick.bind(this);
-    this.getVScodeLink = this.getVScodeLink.bind(this);
   }
 
   handleBackButtonClick() {
```
Comment (msnehagup) [inline ServerlessLibraryAPI/ClientApp/src/components/DetailView/DetailView.js]: handleBackButtonClick is not needed anymore in this component

Comment (msnehagup) [inline ServerlessLibraryAPI/ClientApp/src/components/DetailView/DetailView.js]: At line 28: this.setState({ repository: currentItem.repository });
I am not able to see 'repository' in the state

File: ServerlessLibraryAPI/ClientApp/src/components/DetailView/DetailPageContent.js [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+  }
+});
+
+class DetailPageContent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      armTemplateText: "",
+      markdownText: "",
+      selectedKey: "licence"
```
[the remark this answers is not in the document]
Comment (pravagar) [reply ServerlessLibraryAPI/ClientApp/src/components/DetailView/DetailPageContent.js]: You are right .That will be done once final UX comes. As of now I don't want to pull gitgub resources unnecessarily

File: ServerlessLibraryAPI/ClientApp/src/components/DetailView/DetailPageContent.js (hunk 1 above)
Comment (pravagar) [reply ServerlessLibraryAPI/ClientApp/src/components/DetailView/DetailPageContent.js]: will address this later once we freeze on the behaviour for error handling

**Pull request merged**